read-fonts/src/tables/ift.rs

```diff
@@ -0,0 +1,344 @@
+//! Incremental Font Transfer [Patch Map](https://w3c.github.io/IFT/Overview.html#font-format-extensions)
+
+include!("../../generated/generated_ift.rs");
+
+use core::str::Utf8Error;
+use std::{ops::RangeInclusive, str};
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct U8Or16(u16);
+
+impl ReadArgs for U8Or16 {
+    type Args = u16;
+}
+
+impl ComputeSize for U8Or16 {
+    fn compute_size(max_entry_index: &u16) -> Result<usize, ReadError> {
+        Ok(if *max_entry_index < 256 { 1 } else { 2 })
+    }
+}
+
+impl FontReadWithArgs<'_> for U8Or16 {
+    fn read_with_args(data: FontData<'_>, max_entry_index: &Self::Args) -> Result<Self, ReadError> {
+        if *max_entry_index < 256 {
+            data.read_at::<u8>(0).map(|v| Self(v as u16))
+        } else {
+            data.read_at::<u16>(0).map(Self)
+        }
+    }
+}
+
+impl U8Or16 {
+    #[inline]
+    pub fn get(self) -> u16 {
+        self.0
+    }
+}
+
+impl<'a> PatchMapFormat1<'a> {
+    pub fn get_compatibility_id(&self) -> [u32; 4] {
+        [
+            self.compatibility_id().first().unwrap().get(),
+            self.compatibility_id().get(1).unwrap().get(),
+            self.compatibility_id().get(2).unwrap().get(),
+            self.compatibility_id().get(3).unwrap().get(),
+        ]
+    }
+
+    pub fn gid_to_entry_iter(&'a self) -> impl Iterator<Item = (GlyphId, u16)> + 'a {
+        GidToEntryIter {
+            glyph_map: self.glyph_map().ok(),
+            glyph_count: self.glyph_count(),
+            gid: self
+                .glyph_map()
+                .map(|glyph_map| glyph_map.first_mapped_glyph() as u32)
+                .unwrap_or(0),
+        }
+        .filter(|(_, entry_index)| *entry_index > 0)
+    }
+
+    pub fn entry_count(&self) -> u32 {
+        self.max_entry_index() as u32 + 1
+    }
+
+    pub fn uri_template_as_string(&self) -> Result<&str, Utf8Error> {
```

Comment:
I'm fine with leaving this as `Utf8Error` but mapping it to a `ReadError` variant (maybe `MalformedData("URI template not valid UTF-8")`) might make this API easier to use.